odoo/required.py

```diff
@@ -0,0 +1,248 @@
+from collections.abc import Mapping
```

Comment:
Do not introduce a new top-level module. Things in here should stay part of `odoo.tools`.